genus/src/main/scala/co/topl/genus/interpreters/MongoChainHeight.scala

```
@@ -63,7 +64,9 @@ object MongoChainHeight {
             .toSeq
         )
         // track the latest value seen
-        .toMat(TrackLatestFlow.create[BlockHeight].to(Sink.ignore))(Keep.right)
+        .viaMat(TrackLatestFlow.create[BlockHeight])(Keep.right)
+        // ignore value after tracked
+        .toMat(Sink.ignore)(Keep.left)
```

You _could_ consider keeping both materialized values here.  The `left` is a function that returns a Future, and the `right` is just a Future.  So you could do something like `() => Future.firstCompletedOf(rightFuture, leftFunction())`.  That way, once the stream completes, these function calls will immediately fail instead of returning stale data.